production/helm/loki/templates/deployment.yaml

```diff
@@ -63,6 +63,11 @@ spec:
             {{- toYaml .Values.resources | nindent 12 }}
           securityContext:
             readOnlyRootFilesystem: true
+          {{- if .Values.traceEnabled }}
+          env:
```

Comment:
if in the future we want to template more env var we won't be able.

I'd suggest you move this down after env: also I think having a check only on if  .Values.jaegerAgentHost is set is enough, don't think we need `.Values.traceEnabled`